pkg/dao/repository_config.go

```diff
@@ -393,9 +394,11 @@ func (r repositoryConfigDaoImpl) ValidateParameters(orgId string, params api.Rep
 		if err != nil {
 			return response, err
 		} else if response.URL.Valid {
-			repomd := r.validateMetadataPresence(url, &response)
+			repo := r.yumRepo.Repository()
+			repo.URL = url
+			r.validateMetadataPresence(&response)
 			if response.URL.MetadataPresent {
-				r.checkSignaturePresent(&params, repomd, &response)
+				r.checkSignaturePresent(&params, &response)
```

Comment:
For the purposes of mocking, it seems I have to dependency inject the YumRepository interface and also return the Repository struct with the `Repository()` method. 

This seems more cumbersome than what we had before, so open to other ideas.

Reply:
Resolved this with yumRepo.Configure(settings yum.YummySettings) method.